diff.go

```diff
@@ -238,7 +254,11 @@ func idstring(v interface{}) string {
 	case int:
 		return strconv.Itoa(v.(int))
 	default:
-		return fmt.Sprint(v)
+		b, err := msgpack.Marshal(v)
```

Comment:
For the sake of preserving the current output format for other users, I think we need to hide this behind an option to the differ

Reply:
Honestly, I sort of saw this one coming. It was a trade off between keeping backwards compatibility and keeping the output of diff compatible with patch by default.

I did / do however agree with the ask and think we should protect downstream compatibility, breaking users existing code will probably have them walking away.

I put the marshaling behind the StructMapKeySupport option. This however increased complexity enough to drop the unit tests below 80% so Ive done some additional due-diligence and  shored up the tests by adding some coverage for uint (which was the only set of code at 0%)  bringing us back up to 80% coverage across the file set.